pkg/controller/elasticsearch/client/base.go

```diff
@@ -66,10 +69,15 @@ func (c *baseClient) doRequest(context context.Context, request *http.Request) (
 
 	response, err := c.HTTP.Do(withContext)
 	if err != nil {
-		return response, err
+		return response, newDecoratedHTTPError(request, err)
+	}
```

Comment:
unrelated to that PR, but since we're at it: I'm wondering if we should debug log all the requests we make to Elasticsearch here (with their path, not their body).

Reply:
👍 , I'm just tempted to create a dedicated issue. It requires a small change to the current code in order to have the name of the Elasticsearch resource in the log message, which sounds like a "good first issue" for someone who want to familiarize itself with the client code.